tests/integ/modin/frame/test_cache_result.py

```diff
@@ -125,35 +142,135 @@ def test_cache_result_dataframe_complex_correctness(
-
-        if not inplace:
-            # If inplace, then cached_snow_df IS snow_df, so this check
-            # is meaningless.
-            with SqlCounter(query_count=1):
-                assert_snowpark_pandas_equals_to_pandas_without_dtypecheck(
-                    snow_df, native_df
-                )
 
         with SqlCounter(query_count=1):
             assert_snowpark_pandas_equals_to_pandas_without_dtypecheck(
                 cached_snow_df, native_df
             )
 
-        assert cached_time < uncached_time
+    def test_cache_result_post_pivot(self, inplace, simple_test_data):
+        pivot_kwargs = {
+            "index": "col1",
+            "columns": ["col0", "col1"],
+            "values": "col2",
+            "aggfunc": ["mean", "max"],
+        }
+        snow_df = pd.DataFrame(simple_test_data).pivot_table(**pivot_kwargs)
+        native_df = native_pd.DataFrame(simple_test_data)
+        native_df = perform_chained_operations(
+            native_df.pivot_table(**pivot_kwargs), native_pd
+        )
+        with SqlCounter(query_count=1, join_count=10, union_count=9):
+            snow_df = perform_chained_operations(snow_df, pd)
+            assert_snowpark_pandas_equals_to_pandas_without_dtypecheck(
+                snow_df, native_df
+            )
+
+        with SqlCounter(query_count=1, join_count=1):
+            snow_df = pd.DataFrame(simple_test_data).pivot_table(**pivot_kwargs)
+            cached_snow_df = cache_and_return_df(snow_df, inplace)
+
+        with SqlCounter(query_count=1, union_count=9):
+            cached_snow_df = perform_chained_operations(cached_snow_df, pd)
+            assert_snowpark_pandas_equals_to_pandas_without_dtypecheck(
+                cached_snow_df, native_df
+            )
+
+    def test_cache_result_post_apply(self, inplace, simple_test_data):
+        native_df = perform_chained_operations(
+            native_pd.DataFrame(simple_test_data).apply(lambda x: x + x, axis=1),
+            native_pd,
+        )
+        with SqlCounter(query_count=6, union_count=9, udtf_count=1):
+            snow_df = pd.DataFrame(simple_test_data).apply(lambda x: x + x, axis=1)
+            repr(snow_df)
+            snow_df = perform_chained_operations(snow_df, pd)
+            assert_snowpark_pandas_equals_to_pandas_without_dtypecheck(
+                snow_df, native_df
+            )
+
+        with SqlCounter(query_count=5, udtf_count=1):
+            snow_df = pd.DataFrame(simple_test_data).apply(lambda x: x + x, axis=1)
+            cached_snow_df = cache_and_return_df(snow_df, inplace)
+
+        with SqlCounter(query_count=2, union_count=9):
+            repr(cached_snow_df)
+            cached_snow_df = perform_chained_operations(cached_snow_df, pd)
+            assert_snowpark_pandas_equals_to_pandas_without_dtypecheck(
+                cached_snow_df,
+                native_df,
+            )
+
+    def test_cache_result_post_applymap(self, inplace, simple_test_data):
```

Comment:
I don't think `applymap` does caching of its own. The query overheads in this test case are from setting up the UDF and defining it, so we just end up paying an extra query to cache the UDF result. I would add a comment explaining that.